pyCGM_Single/tests/test_pycgm_utils.py

```diff
@@ -0,0 +1,364 @@
+import pyCGM_Single.pyCGM as pyCGM
+import pytest
+import numpy as np
+
+rounding_precision = 8
+
+class TestUtils():
+    """
+    This class tests the utils functions in pyCGM.py:
+    findwandmarker
+    cross
+    norm2d
+    norm3d
+    normDiv
+    matrixmult
+    rotmat
+    """
+    rand_coor = [np.random.randint(0, 10), np.random.randint(0, 10), np.random.randint(0, 10)]
+
+    @pytest.mark.parametrize(["frame", "thorax", "expected"], [
+        ({'RSHO': [428.88476562, 270.552948, 1500.73010254], 'LSHO': [68.24668121, 269.01049805, 1510.1072998]}, [[[256.23991128535846, 365.30496976939753, 1459.662169500559], rand_coor, rand_coor], [256.149810236564, 364.3090603933987, 1459.6553639290375]], [[255.92550222678443, 364.3226950497605, 1460.6297868417887], [256.42380097331767, 364.27770361353487, 1460.6165849382387]]),
+        ({'RSHO': [0, 0, 1], 'LSHO': [0, 1, 0]}, [[[1, 0, 0], rand_coor, rand_coor], [0, 0, 0]], [[0, 1, 0], [0, 0, 1]]),
+        ({'RSHO': [0, 1, 1], 'LSHO': [1, 1, 1]}, [[[1, 0, 0], rand_coor, rand_coor], [0, 0, 0]], [[0, 0.70710678, -0.70710678], [0, -0.70710678, 0.70710678]]),
+        ({'RSHO': [0, 1, 1], 'LSHO': [1, 1, 1]}, [[[1, 0, 0], rand_coor, rand_coor], [-1, 0, 0]], [[-1, 0.70710678, -0.70710678], [-1, -0.70710678, 0.70710678]]),
+        ({'RSHO': [1, 2, 1], 'LSHO': [2, 1, 2]}, [[[1, 0, 0], rand_coor, rand_coor], [0, 0, 0]], [[0, 0.4472136, -0.89442719], [0, -0.89442719, 0.4472136]]),
+        ({'RSHO': [1, 1, 1], 'LSHO': [1, 1, 1]}, [[[1, 0, 1], rand_coor, rand_coor], [0, 0, 0]], [[0.70710678, 0, -0.70710678], [-0.70710678, 0, 0.70710678]]),
+        ({'RSHO': [1, 1, 1], 'LSHO': [1, 1, 1]}, [[[1, 0, 1], rand_coor, rand_coor], [0, 0, 1]], [[0, 0, 0], [0, 0, 2]]),
+        ({'RSHO': [0, 1, 0], 'LSHO': [0, 0, -1]}, [[[0, 3, 4], rand_coor, rand_coor], [0, 0, 0]], [[1, 0, 0], [-1, 0, 0]]),
+        ({'RSHO': [1, 0, 0], 'LSHO': [0, 1, 0]}, [[[7, 0, 24], rand_coor, rand_coor], [0, 0, 0]], [[0, -1, 0], [-0.96, 0, 0.28]]),
+        ({'RSHO': [1, 0, 0], 'LSHO': [0, 0, 1]}, [[[8, 0, 6], rand_coor, rand_coor], [8, 0, 0]], [[8, 1, 0], [8, -1, 0]])
+    ])
+    def test_findwandmarker(self, frame, thorax, expected):
+        """
+        This test provides coverage of the findwandmarker function in pyCGM.py, defined as findwandmarker(frame,thorax)
+        where frame is a dictionary of x, y, z positions and marker names and thorax is the thorax axis.
+
+        This test takes 3 parameters:
+        frame: dictionary of marker lists
+        thorax: the x,y,z position of the thorax
+        expected: the expected result from calling findwandmarker on frame and thorax. This results is the wand marker
+        position used for calculating knee joint center later. The wand marker position is returned as a 2x3 array
+        containing the right wand marker x,y,z positions (1x3) followed by the left wand marker x,y,z positions (1x3).
```

Comment:
Can you expand on what the wand marker is for this test.  I think this function is really the most important one in the file here.